src/main/java/org/mockito/internal/util/reflection/InstrumentationMemberAccessor.java

```diff
@@ -0,0 +1,363 @@
+        }
+    }
+
+    @Override
+    public Object newInstance(Constructor<?> constructor, Object... arguments)
+            throws InstantiationException, InvocationTargetException {
+        if (Modifier.isAbstract(constructor.getDeclaringClass().getModifiers())) {
+            throw new InstantiationException(
+                    "Cannot instantiate abstract " + constructor.getDeclaringClass().getTypeName());
+        }
+        assureArguments(constructor, null, null, arguments, constructor.getParameterTypes());
+        try {
+            Object module = getModule.bindTo(constructor.getDeclaringClass()).invokeWithArguments();
+            String packageName = constructor.getDeclaringClass().getPackage().getName();
+            assureOpen(module, packageName);
+            MethodHandle handle =
+                    ((MethodHandles.Lookup)
+                                    privateLookupIn.invokeExact(
+                                            constructor.getDeclaringClass(),
+                                            DISPATCHER.getLookup()))
+                            .unreflectConstructor(constructor);
+            try {
+                return handle.invokeWithArguments(arguments);
+            } catch (Throwable t) {
+                throw new InvocationTargetException(t);
+            }
+        } catch (InvocationTargetException e) {
+            throw e;
+        } catch (Throwable t) {
+            throw new IllegalStateException(
+                    "Could not construct "
+                            + constructor
+                            + " with arguments "
+                            + Arrays.toString(arguments),
+                    t);
+        }
+    }
+
+    @Override
+    public Object invoke(Method method, Object target, Object... arguments)
+            throws InvocationTargetException {
+        assureArguments(
+                method,
+                Modifier.isStatic(method.getModifiers()) ? null : target,
+                method.getDeclaringClass(),
+                arguments,
+                method.getParameterTypes());
+        try {
+            Object module = getModule.bindTo(method.getDeclaringClass()).invokeWithArguments();
+            String packageName = method.getDeclaringClass().getPackage().getName();
+            assureOpen(module, packageName);
+            MethodHandle handle =
+                    ((MethodHandles.Lookup)
+                                    privateLookupIn.invokeExact(
+                                            method.getDeclaringClass(), DISPATCHER.getLookup()))
+                            .unreflect(method);
+            if (!Modifier.isStatic(method.getModifiers())) {
+                handle = handle.bindTo(target);
+            }
+            try {
+                return handle.invokeWithArguments(arguments);
+            } catch (Throwable t) {
+                throw new InvocationTargetException(t);
+            }
```

Comment:
There is quite a bit of duplication across these methods. Does it maybe make sense to extract it out? The only difference I am spotting is the `unreflect` in here, where others use for example `unreflectGetter`.

Reply:
I didn't want to factor it out too much since I don't think it makes the code easier to read and it would be harded to extend if you wanted to add a special case for one of the dispatchers at some point that does not apply to the other.